### PR TITLE
Adding formatter support for HLSL and AZSL

### PR DESCRIPTION
**Pull request by xeechou**
[HLSL](https://github.com/xeechou/hlsl-mode.el) is a popular C++ like shading language used in gaming industry. 

[AZSL](https://github.com/xeechou/azsl-mode.el) (Amazon shading language) is a HLSL dialect used in [o3de](https://github.com/o3de/o3de/).

Both languages use very similar syntax to C++. This patch add the two languages support in format-all.

I have implemented both languages major mode in emacs.

### CODE REVIEW

File: format-all.el [hunk 1]
```diff
@@ -1517,6 +1521,8 @@ unofficial languages IDs are prefixed with \"_\"."
       (and (equal major-mode 'ledger-mode) "_Ledger")
       (and (equal major-mode 'nginx-mode) "_Nginx")
       (and (equal major-mode 'snakemake-mode) "_Snakemake")
+      (and (equal major-mode 'hlsl-mode) "_HLSL")
```
Comment (lassik) [inline format-all.el:1524]: Please remove this line.

File: format-all.el [hunk 2]
```diff
@@ -210,7 +212,9 @@
     ("_Gleam" gleam)
     ("_Ledger" ledger-mode)
     ("_Nginx" nginxfmt)
-    ("_Snakemake" snakefmt))
+    ("_Snakemake" snakefmt)
+    ("_HLSL" clang-format)
```
Comment (lassik) [inline format-all.el:216]: Please change `"_HLSL"` to `"HLSL"`.

Comment (lassik) [reply format-all.el:216]: And move `"HLSL"` between HCL and HTML.

File: format-all.el [hunk 3]
```diff
@@ -210,7 +212,9 @@
...
+    ("_AZSL" clang-format))
```
Comment (lassik) [inline format-all.el:217]: Please move `_AZSL` between `_Angular` and `_Beancount`.

File: format-all.el [hunk 4]
```diff
@@ -1517,6 +1521,8 @@ unofficial languages IDs are prefixed with \"_\"."
       (and (equal major-mode 'ledger-mode) "_Ledger")
       (and (equal major-mode 'nginx-mode) "_Nginx")
       (and (equal major-mode 'snakemake-mode) "_Snakemake")
+      (and (equal major-mode 'hlsl-mode) "_HLSL")
+      (and (equal major-mode 'azsl-mode) "_AZSL")
```
Comment (lassik) [inline format-all.el:1525]: Please move `_AZSL` before `_Beancount`.

**Review by lassik — CHANGES_REQUESTED** [review]

**Comment by lassik** [comment]
Thank you.

HLSL is officially recognized by GitHub Linguist, so I [added it to the language-id library](https://github.com/lassik/emacs-language-id/commit/435114f208b97e97aa1576ef1966a33d90cad01b). No `_` needed.

Please also add `"HLSL"` and `"_AZSL"` to the `:languages` list inside `(define-format-all-formatter clang-format ...)`.

**Comment by xeechou** [comment]
Hi, thanks. The MR is updated.

**Comment by lassik** [comment]
Thank you for the good work!

**Pull request merged**